codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/handler/SockJsHandler.java

```diff
@@ -0,0 +1,273 @@
+                                      final TransportType transportType,
+                                      final String sessionId) throws Exception {
+        switch (transportType) {
+        case XHR:
+            addTransportHandler(new XhrPollingTransport(sockJsConfig, request), ctx);
+            addSessionHandler(new PollingSessionState(sessions, session(factory, sessionId)), ctx);
+            break;
+        case JSONP:
+            addTransportHandler(new JsonpPollingTransportInbound(), ctx);
+            addTransportHandler(new JsonpPollingTransportOutbound(sockJsConfig, request), ctx);
+            addSessionHandler(new PollingSessionState(sessions, session(factory, sessionId)), ctx);
+            break;
+        case XHR_SEND:
+            checkSessionExists(sessionId, request);
+            addTransportHandler(new XhrSendTransport(sockJsConfig), ctx);
+            addSessionHandler(new SendingSessionState(sessions, sessions.get(sessionId)), ctx);
+            break;
+        case XHR_STREAMING:
+            addTransportHandler(new XhrStreamingTransport(sockJsConfig, request), ctx);
+            addSessionHandler(new StreamingSessionState(sessions, session(factory, sessionId)), ctx);
+            break;
+        case EVENTSOURCE:
+            addTransportHandler(new EventSourceTransport(sockJsConfig, request), ctx);
+            addSessionHandler(new StreamingSessionState(sessions, session(factory, sessionId)), ctx);
+            break;
+        case HTMLFILE:
+            addTransportHandler(new HtmlFileTransportInbound(), ctx);
+            addTransportHandler(new HtmlFileTransportOutbound(sockJsConfig, request), ctx);
+            addSessionHandler(new StreamingSessionState(sessions, session(factory, sessionId)), ctx);
+            break;
+        case JSONP_SEND:
+            checkSessionExists(sessionId, request);
+            addTransportHandler(new JsonpSendTransport(sockJsConfig), ctx);
+            addSessionHandler(new SendingSessionState(sessions, sessions.get(sessionId)), ctx);
+            break;
+        case WEBSOCKET:
+            addTransportHandler(new WebSocketTransport(sockJsConfig), ctx);
+            addSessionHandler(new WebSocketSessionState(newSession(factory)), ctx);
+            break;
+        }
+        ctx.fireChannelRead(request);
+    }
+
+    private static void addTransportHandler(final ChannelHandler transportHandler, final ChannelHandlerContext ctx) {
+        ctx.pipeline().addLast(transportHandler);
+    }
+
+    private static void addSessionHandler(final SessionState sessionState, final ChannelHandlerContext ctx) {
+        ctx.pipeline().addLast(new SessionHandler(sessionState));
+    }
+
+    private static void checkSessionExists(final String sessionId, final HttpRequest request)
+            throws SessionNotFoundException {
+        if (!sessions.containsKey(sessionId)) {
+            throw new SessionNotFoundException(sessionId, request);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)